File: src/pattern/factory/java/IDCardFactory.java
package pattern.factory.java;

import java.util.ArrayList;
import java.util.List;

public class IDCardFactory extends Factory {
    private List owners = new ArrayList<String>();

    @Override
    protected Product createProduct(String name) {
        return new IDCard(name);
    }

    @Override
    protected void registerProduct(Product product) {
        owners.add( ((IDCard)product).getOwner());
    }

    public List getOwners(){
        return owners;
    }
}

File: src/pattern/factory/kotlin/IDCardFactory.kt
package pattern.factory.kotlin

class IDCardFactory: Factory(){
    var owners = arrayListOf<IDCard>()

    override fun createProduct(name: String): Product {
        return IDCard(name)
    }

    override fun registerProduct(product: Product) {
        owners.add(product as IDCard)
    }
}